src/app/core/testing/utils/mock-storage-generator.utils.ts

```diff
@@ -0,0 +1,381 @@
+import { PoolTopologyCategory } from 'app/enums/pool-topology-category.enum';
+import { TopologyItemType } from 'app/enums/v-dev-type.enum';
+import { PoolInstance } from 'app/interfaces/pool.interface';
+import {
+  StorageDashboardDisk,
+  TopologyDisk,
+  TopologyItem,
+  TopologyItemStats,
+  VDev,
+} from 'app/interfaces/storage.interface';
+
+export enum MockStorageScenario {
+  Default = 'default',
+  Uniform = 'uniform',
+  Multi = 'multi', // Multiple problems
+  MixedVdevLayout = 'mixedVdevLayout',
+  MixedVdevCapacity = 'mixedVdevCapacity',
+  MixedDiskCapacity = 'mixedDiskCapacity',
+  MixedVdevWidth = 'mixedVdevWidth',
+  NoRedundancy = 'noRedundancy',
+}
+
+export interface MockStorage {
+  poolState: PoolInstance;
+  disks: StorageDashboardDisk[];
+}
+
+export interface MockTopology {
+  topologyItems: TopologyItem[];
+  disks: StorageDashboardDisk[];
+}
+
+export class MockStorageGenerator {
+  poolState: PoolInstance;
+  disks: StorageDashboardDisk[];
+
+  constructor() {
+    // Creates a pool with empty topologies
+    const storage = this.generateStorage();
+    this.poolState = storage.poolState;
+    this.disks = storage.disks;
+  }
+
+  // Generate Empty Topology & Disks
+  private generateStorage(): MockStorage {
+    const pool = {
+      id: 1,
+      name: 'MOCK_POOL',
+      healthy: true,
+      status: 'ONLINE',
+      topology: {
+        data: [],
+        special: [],
+        log: [],
+        spare: [],
+        cache: [],
+        dedup: [],
+      },
+    } as unknown as PoolInstance;
+
+    const disks: StorageDashboardDisk[] = [];
+
+    return { poolState: pool, disks };
+  }
+
+  addDataTopology(scenario: MockStorageScenario,
```

Comment:
If you have more than 3 arguments, it's better to just have one argument called `options`.